### PR TITLE
Add aws_rds_iam_token utlity module for creating rds iam tokens for authenticating over IAM towards RDS or Aurora

### PR DESCRIPTION
**Pull request by onno-vos-dev**
Based off of: [rds/auth/connect.go](https://github.com/aws/aws-sdk-go-v2/blob/main/feature/rds/auth/connect.go)

Will port this to aws-elixir somewhere in the coming days.

### CODE REVIEW

**Review by mikpe — APPROVED** [review]

File: src/aws_rds_iam_token.erl [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+-export([create/4]).
+
+-define(SIGNING_ID, <<"rds-db">>).
+-define(EMPTY_PAYLOAD_HASH, <<"e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855">>).
+
+%%====================================================================
+%% API
+%%====================================================================
+-spec create(aws_client:client(), binary(), non_neg_integer(), binary()) -> {ok, binary()}.
+create(Client, DBEndpoint, DBPort, DBUser) when is_map(Client) andalso
```
Comment (jadeallenx) [inline src/aws_rds_iam_token.erl:11]: Function name spaces are totally flat in Erlang, so this is not a good function name to export. I would strongly recommend you change this to `rds_token_create` or something like that.

**Review by jadeallenx — CHANGES_REQUESTED** [review]

**Review by jesperes — APPROVED** [review]

**Review by mikpe — APPROVED** [review]

**Comment by onno-vos-dev** [comment]
Rebased and squashed. Merging and rolling out a release as soon as the build is green 👍

**Comment by onno-vos-dev** [comment]
2 out of 4 builds failed due to flakyness with Github uploading of the docs. Merging anyway.

File: README.md [hunk 2]
```diff
@@ -53,6 +53,18 @@ through the `aws_s3_presigned_url` module.
 #{<<"HasMoreStreams">> => false,<<"StreamNames">> => []}
 ```
 
+### AWS RDS IAM Token Creation
+Support for creating IAM Tokens (more info [here](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.Connecting.html)) has been added as part of the `aws_rds_iam_token` module.
+This allows for easy creation of RDS/Aurora tokens to be used for IAM based authentication instead of username/password combination.
+```erlang
+> Client = aws_client:make_temporary_client(<<"AccessKeyID">>, <<"SecretAccessKey">>, <<"Token">>, <<"eu-west-1">>).
+[...]
+> {ok, Url} = create(Client, <<"db_endpoint">>, 5432, <<"db_user">>).
```
Comment (jadeallenx) [inline README.md:62]: The docs are out of date because of the function name change :)

Comment (onno-vos-dev) [reply README.md:62]: Already fixed in master 👍 I noticed too so followed up on it knstead

**Comment by jadeallenx** [comment]
Awesome work, thanks!

**Pull request merged**